bin/tx-prover/src/proxy/mod.rs

```diff
@@ -0,0 +1,193 @@
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use once_cell::sync::Lazy;
+use pingora::{
+    http::ResponseHeader,
+    lb::Backend,
+    prelude::*,
+    upstreams::peer::{Peer, ALPN},
+};
+use pingora_core::{upstreams::peer::HttpPeer, Result};
+use pingora_limits::rate::Rate;
+use pingora_proxy::{ProxyHttp, Session};
+use tokio::sync::RwLock;
+use tracing::error;
+
+/// Timeout for the requests
+const TIMEOUT_SECS: Option<Duration> = Some(Duration::from_secs(100));
+/// Maximum number of items per queue
+const MAX_QUEUE_ITEMS: usize = 10;
+
+/// Load balancer that uses a round robin strategy
+pub struct LB(Arc<LoadBalancer<RoundRobin>>);
+
+impl LB {
+    pub fn new(upstreams: LoadBalancer<RoundRobin>) -> Self {
+        Self(Arc::new(upstreams))
+    }
+}
+
+/// Rate limiter
+static RATE_LIMITER: Lazy<Rate> = Lazy::new(|| Rate::new(Duration::from_secs(1)));
+
+/// Maximum amount of request per second per client
+static MAX_REQ_PER_SEC: isize = 5;
+
+/// Shared state. It is a map of backends to a vector of request IDs
+static QUEUES: Lazy<RwLock<HashMap<Backend, Vec<String>>>> =
```

Comment:
Is there any benefit to these being `Lazy<_>`? As opposed to just being initialized

Reply:
The issue is that `HashMap::new` is non-const, is there any other way to avoid the Lazy?

Reply:
Ah for some reason I thought it would be fine just looking at the `RwLock`, but didn't realize the hashmap would be the problem!